journal.py

```diff
@@ -1,128 +1,131 @@
-INSERT INTO entries (title, text, created) VALUES (%s, %s, %s)
-"""
-DB_ENTRIES_LIST = """
-SELECT id, title, text, created FROM entries ORDER BY created DESC
-"""
-UPDATE_ENTRY = """
-UPDATE entries SET (title, text, created) = (%s, %s, %s) WHERE id = %s
-"""
-READ_ENTRY = """
-SELECT id, title, text, created FROM entries WHERE id = %s
-"""
-logging.basicConfig()
-log = logging.getLogger(__file__)
-
 
-# @view_config(route_name='home', renderer='string')
-# def home(request):
-#     return "Hello World"
+here = os.path.dirname(os.path.abspath(__file__))
 
 
-def connect_db(settings):
-    """Return a connection to the configured database"""
-    return psycopg2.connect(settings['db'])
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+Base = declarative_base()
 
 
-def init_db():
-    """Create database dables defined by DB_SCHEMA
-    Warning: This function will not update existing table definitions
-    """
-    settings = {}
-    settings['db'] = os.environ.get(
-        'DATABASE_URL', 'dbname=learning_journal user=jakeanderson'
+class Entry(Base):
+    __tablename__ = 'entries'
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
     )
-    with closing(connect_db(settings)) as db:
-        db.cursor().execute(DB_SCHEMA)
-        db.commit()
-
 
-@subscriber(NewRequest)
-def open_connection(event):
-    request = event.request
-    settings = request.registry.settings
-    request.db = connect_db(settings)
-    request.add_finished_callback(close_connection)
+    def __repr__(self):
+        return u"{}: {}".format(self.__class__.__name__, self.title)
+
+    @classmethod
+    def all(cls):
+        return DBSession.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def by_id(cls, id):
+        return DBSession.query(cls).filter(cls.id == id).one()
+
+    @classmethod
+    def from_request(cls, request):
+        title = request.params.get('title', None)
+        text = request.params.get('text', None)
+        created = datetime.datetime.utcnow()
+        new_entry = cls(title=title, text=text, created=created)
+        DBSession.add(new_entry)
+
+    @classmethod
+    def desc_new(cls):
+        return DBSession.query(cls).order_by(cls.created.desc()).first()
+
+    def json_detail(self):
+        '''Displays a JSON object for detailed view'''
+        return {'title': self.title,
+                'text': self.render_markdown(),
+                'created': self.created.strftime('%b %d, %Y'),
+                'id': self.id}
+
+    def json_edit(self):
+        '''Does not show HTML during editing'''
+        return {'title': self.title,
+                'text': self.text,
+                'created': self.created.strftime('%b %d, %Y'),
+                'id': self.id}
```

Comment:
I'd rather these were one method with a switch that determined whether to render the text to markdown or not.  This is a bit too much repetition for my taste.